x-pack/plugins/security_solution/public/resolver/view/clickthrough.test.tsx

```diff
@@ -73,6 +73,32 @@ describe('Resolver, when analyzing a tree that has no ancestors and 2 children',
       });
     });
 
+    it('should render 3 elements with "treeitem" roles, each owned by an element with a "tree" role', async () => {
+      await expect(
+        simulator.map(() => ({
+          nodesOwnedByTrees: simulator.testSubject('resolver:node').filterWhere((domNode) => {
+            /**
+             * This test verifies corectness w.r.t. the tree/treeitem roles
+             * From W3C: `Authors MUST ensure elements with role treeitem are contained in, or owned by, an element with the role group or tree.`
+             *
+             * https://www.w3.org/TR/wai-aria-1.1/#tree
+             * https://www.w3.org/TR/wai-aria-1.1/#treeitem
+             *
+             * w3c defines two ways for an element to be an "owned element"
+             *  1. Any DOM descendant
+             *  2. Any element specified as a child via aria-owns
+             *  (see: https://www.w3.org/TR/wai-aria-1.1/#dfn-owned-element)
+             *
+             * In the context of Resolver (as of this writing) nodes/treeitems are children of the tree,
+             * but they could be moved out of the tree, provided that the tree is given an `aria-owns`
+             * attribute referring to them (method 2 above).
+             */
+            return domNode.closest('[role="tree"]').length === 1;
```

Comment:
Would it be worth making it `databaseDocumentID` specific so `xyz-tree`. Just thinking ahead of the multi-resolver scenario? This would catch any bugs as is, but might be better to use a more specific selector?

Reply:
I think `closest` supports the assertion for each one, regardless of how many other resolver are/aren't currently rendered. What I'm trying to test here is that the `treeitem` has one `tree` that it's contained in. That should hold, I think, even if you had something odd like resolvers embedded in other resolvers. I think you'd have to bring the documentID into play if you used the second technique (aria-owns), but I just left a long comment so hopefully someone will understand that if they ever use that technique.

Reply:
Okay, yea. I guess I'm thinking more about which tree it belongs to rather than it just belonging to one tree